storage/kafka/src/main/java/io/apicurio/registry/kafka/KafkaRegistryStorage.java

```diff
@@ -48,8 +59,16 @@
 @ApplicationScoped
 public class KafkaRegistryStorage extends SimpleMapRegistryStorage implements KafkaRegistryStorageHandle {
 
+    private static final Logger log = LoggerFactory.getLogger(KafkaRegistryStorage.class);
+
+    @ConfigProperty(name = "registry.kafka.snapshot.requests", defaultValue = "1000")
+    long snapshotRequests;
```

Comment:
I'll add a checker thread, which will check for autosnapshot period -- so the snapshot is not too old wrt retention (aka if things are configured properly, we should not loose any state).

Reply:
> I know you want to keep configuration of the topics out of this part of the code, but details like this are important for guaranteeing the functionality.

Hmmm, I guess this cannot be done w/o KafkaAdmin usage?

Reply:
That's correct (assuming you mean `AdminClient`), but don't let the `AdminClient` name put you off. It's not bad practice to use it for producing/consuming application if they need to, for example, create topics with configs which differ from the defaults (it's used by Kafka Streams for this purpose, for example). It's not _just_ meant for 'admin' usage.

Reply:
OK
Could you then help me with some  of the default settings?

I guess something like this:
- snapshot topic should be compact
- snapshot period check should be N-times smaller then registry topic retention
- ...

Reply:
I think those are the only things you should be checking and enforcing. The only other thing to consider is the number of partitions. If the number of partitions had to be increased then consumers would start observing different keys. I don't think that's a problem for what you're doing here though.

Reply:
Atm, I have a single partition - to re-use offset as global id.

Reply:
If you have a single partition then the consumers in your consumer group won't really function in a distributed way (the group leader assigns each partition to a single consumer, "spare" consumers are basically hot standbys in case an active consumer fails). From the PR it's not completely clear to me whether you're expecting the consumers in each registry instance to be in the same group. If your design requires they be in the same group then this is a problem.

Since we're not expecting the topic to have significant throughput it's probably OK from a scalability PoV.

Reply:
> * snapshot period check should be N-times smaller then registry topic retention

If they're both time durations doesn't that mean that if the registry is not able to snapshot (i.e. is not deployed) for N * topic duration then we lose data?

Reply:
> From the PR it's not completely clear to me whether you're expecting the consumers in each registry instance to be in the same group.

No, they should explicitly be in different group -- so all nodes are consumers.
Atm I simply generate UUID, if no group.id is set.

Reply:
> If they're both time durations doesn't that mean that if the registry is not able to snapshot (i.e. is not deployed) for N * topic duration then we lose data?

Yes, if there wouldn't be snapshot for a long time (more than retention time), we would loose it all.
Hence the retention should not be too small, plus somebody should be monitoring registry.